### PR TITLE
Fix issues with size header in cloning sometimes reading beginning of…

### PR DESCRIPTION
**Pull request by awels**
… data.

Signed-off-by: Alexander Wels <awels@redhat.com>

**What this PR does / why we need it**:
Fixes cloning sometimes missing the first block when cloning.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/cdi-cloner/clonertarget.go [hunk 1]
```diff
@@ -145,18 +145,19 @@ func (r *prometheusProgressReader) updateProgress() {
...
-		return -1
+		return -1, err
+	}
+	if n != len(b) {
+		// Didn't read all 8 bytes..
+		return -1, errors.New("Didn't read all bytes for size header")
 	}
-	klog.V(1).Infoln(fmt.Sprintf("total size: %s", totalText))
-	return total
+	size := int64(binary.LittleEndian.Uint64(b))
```
Comment (aglitke) [inline cmd/cdi-cloner/clonertarget.go]: Is 8 bytes enough to hold the maximum size you want to support?  The largest uint64 is 18446744073709551616 which would take 20 ascii bytes to represent.  

Also, You're parsing a 64 bit unsigned integer and converting it to a 64 bit signed integer.  Aren't you losing half of your precision?  Will the size 2^63+1 (9223372036854775809) be handled correctly?

File: tests/framework/pv.go [hunk 2]
```diff
@@ -15,3 +15,28 @@ func (f *Framework) CreatePVFromDefinition(def *k8sv1.PersistentVolume) (*k8sv1.
 func (f *Framework) WaitTimeoutForPVReady(pvName string, timeout time.Duration) error {
 	return utils.WaitTimeoutForPVReady(f.K8sClient, pvName, timeout)
 }
+
+// ResetBlockPVtoZero resets the device to the initial state and wipes any junk left on it.
+func (f *Framework) ResetBlockPVtoZero() error {
```
Comment (aglitke) [inline tests/framework/pv.go]: How about naming this clearBlockPV?

File: cmd/cdi-cloner/clonertarget.go (hunk 1 above)
Comment (aglitke) [reply cmd/cdi-cloner/clonertarget.go]: Best we add some testing to this code that to ensure that numbers on the periphery work as expected.  ie. 2^64, 2^32+1, 0, -1, etc.

Comment (awels) [reply cmd/cdi-cloner/clonertarget.go]: int64 by definition is 8 bytes 8x8=64, I am not sending ascii characters, but the actual bytes, so 8 bytes is enough. Yeah there is no reason I can't use uint64 all the way, let me fix that.

Adding tests to catch edge cases.

File: tests/framework/pv.go (hunk 2 above)
Comment (awels) [reply tests/framework/pv.go]: Sure

**Comment by awels** [comment]
ci test please

**Comment by awels** [comment]
ci test please

File: cmd/cdi-cloner/header/sizeheader.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	out, err := os.OpenFile(*namedPipe, os.O_WRONLY, os.ModeNamedPipe)
+	if err != nil {
+		klog.Errorf("Unable to open named pipe: %+v", err)
+		os.Exit(1)
+	}
+	defer out.Close()
+
+	b := make([]byte, 8)
+	binary.LittleEndian.PutUint64(b, size)
+	out.Write(b)
```
Comment (mhenriks) [inline cmd/cdi-cloner/header/sizeheader.go]: I think it would be wise to check the values returned by this function

Comment (awels) [reply cmd/cdi-cloner/header/sizeheader.go]: Switched to writing with print, so this whole file goes away.

**Comment by awels** [comment]
> Wondering why the go program is necessary? Can't we do this in bash?
> 
> On the source, do: `printf "%016x" $size >$pipe_dir`
> 
> On the target, make a string from the first 16 bytes and `strconv.ParseUint(zeroPaddedString, 16, 64)`

Was trying to get everything in 8 bytes instead of 16, but this is much simpler.

File: tests/cloner_test.go [hunk 4]
```diff
@@ -32,7 +32,7 @@ const (
 	controllerSkipPVCCompleteTimeout = 90 * time.Second
 )
 
-var _ = Describe("[rfe_id:1277][crit:high][vendor:cnv-qe@redhat.com][level:component]Cloner Test Suite", func() {
+var _ = FDescribe("[rfe_id:1277][crit:high][vendor:cnv-qe@redhat.com][level:component]Cloner Test Suite", func() {
```
Comment (mhenriks) [inline tests/cloner_test.go]: FDescribe
